firedrake/slope_limiter/limiter.py

```diff
@@ -0,0 +1,121 @@
+from firedrake import *
+from abc import *
+
+class _Limiter(object):
```

Comment:
I anticipate that users will want to inherit from this.  So let's make a it a public class (no `_` to start the name).

Also, you can provide an `__init__` method that has a default signature.  Additionally, a docstring describing what this object is for would be good.